### PR TITLE
Add Namespace Patching to RBAC good practice

### PR DESCRIPTION
**Pull request by smarticu5**
Updating the RBAC Good Practices guide to include information on Patch operations against Namespaces being a security risk.

### CODE REVIEW

File: content/en/docs/concepts/security/rbac-good-practices.md [hunk 1]
```diff
@@ -181,6 +181,13 @@ Users with control over `validatingwebhookconfigurations` or `mutatingwebhookcon
 can control webhooks that can read any object admitted to the cluster, and in the case of
 mutating webhooks, also mutate admitted objects.
 
+### Namespace modification
+
+Users who can perform `patch` operations on `namespace` objects through a namespaced rolebinding can modify 
+labels on that namespace. In clusters where Pod Security Admission is used, this may allow a user to configure the namespace
+for a more permissive policy than intended by the administrators. For clusters where network policy is used, users may be 
+able to gain access to services intended to be blocked.
```
Comment (sftim) [inline content/en/docs/concepts/security/rbac-good-practices.md:189]:
```suggestion
for a more permissive policy than intended by the administrators.
For clusters where NetworkPolicy is used, users may be set labels that indirectly allow
access to services that an administrator did not intend to allow.
```
NetworkPolicy never configures deny rules

**Review by sftim — COMMENTED** [review]
My take. The style nits are important so that people can tell the concept of “services” apart from the Service API.

File: content/en/docs/concepts/security/rbac-good-practices.md [hunk 2]
```diff
@@ -181,6 +181,13 @@ Users with control over `validatingwebhookconfigurations` or `mutatingwebhookcon
 can control webhooks that can read any object admitted to the cluster, and in the case of
 mutating webhooks, also mutate admitted objects.
 
+### Namespace modification
+
+Users who can perform `patch` operations on `namespace` objects through a namespaced rolebinding can modify
```
[the remark this answers is not in the document]
Comment (smarticu5) [reply content/en/docs/concepts/security/rbac-good-practices.md:186]: You don't need to use a ClusterRole. My main concern with this entire concept is that you can create a RoleBinding to a Role which only allows PATCH on a namespce, then use that to change the labels on that ns "from within".

Agreed on your other nits though.

**Comment by sftim** [comment]
/hold

OK to unhold once the reviews I mentioned in https://github.com/kubernetes/website/pull/45664#discussion_r1539282769 are done.

File: content/en/docs/concepts/security/rbac-good-practices.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (sftim) [reply content/en/docs/concepts/security/rbac-good-practices.md:186]: Try this suggestion if you don't want to mention a ClusterRole:

```suggestion
Users who can perform **patch** operations on Namespace objects (through a namespaced RoleBinding to a Role with that access) can modify
```

**Comment by sftim** [comment]
/hold cancel
Thanks @IanColdwater

**Comment by sftim** [comment]
I'm going to use my maintainer access to fix just the style nits - see https://github.com/kubernetes/website/pull/45664/files#r1539294992

**Review by sftim — COMMENTED** [review]
/lgtm
/approve

Drawing on https://github.com/kubernetes/website/pull/45664#event-12268584848 (LGTM from review by @IanColdwater)

**Comment by tabbysable** [comment]
Thanks so much, @smarticu5, for the research here and the followup to call attention to this surprising behavior.

/lgtm

/honk

**Comment by smarticu5** [comment]
Appreciate this is closed now, thanks all. I've added a PoC using a minimal role and rolebinding to my own repo (https://github.com/smarticu5/kubernetes_manifests/tree/main/pocs/namespace-labelling) for future reference.

**Pull request merged**